Source/SteveSloth/Public/MyPlayer.h

```diff
@@ -112,6 +112,24 @@ class STEVESLOTH_API AMyPlayer : public ACharacter
 	UPROPERTY(EditAnywhere, Category = "Input|Actions")
 	UInputAction* PAiming;
 
+	UPROPERTY(EditAnywhere, Category = "Input|Actions")
+	UInputAction* PGrappplingHook;
+
+	UPROPERTY(EditAnywhere, Category = "Input|Actions")
+	UInputAction* PClimbingClaw;
+
+	UPROPERTY(EditAnywhere, Category = "Input|Actions")
+	UInputAction* PShovel;
+
+	UPROPERTY(EditAnywhere, Category = "Input|Actions")
+	UInputAction* PPropeller;
+
+	UPROPERTY(EditAnywhere, Category = "Input|Actions")
+	UInputAction* PMagnetic;
```

Comment:
we are never going to use all the functions at the same time. So the functions should not have their own input, should just be one input with a switch state inside it that calls the function based off of which upgrade is being used at that time.